### PR TITLE
[MIG] donation_base: migration to 13.0

### PR DESCRIPTION
**Pull request by albariera**
No major changes. I have followed the standard guide.

@etobella

### CODE REVIEW

**Review by etobella — APPROVED** [review]
LGTM

**Comment by etobella** [comment]
@OCA/donation-maintainers

File: donation_base/security/tax_receipt_security.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  © 2014-2016 Akretion France (Alexis de Lattre <alexis.delattre@akretion.com>)
+  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+-->
+<odoo noupdate="1">
+    <record id="donation_tax_receipt_company_rule" model="ir.rule">
+        <field name="name">Donation Tax Receipt Multi-company</field>
+        <field name="model_id" ref="model_donation_tax_receipt" />
+        <field
+            name="domain_force"
+        >['|', ('company_id', '=', False), ('company_id', 'in', company_ids)]</field>
```
Comment (tbaden) [inline donation_base/security/tax_receipt_security.xml:13]: why was this changed from child_of to in?

File: donation_base/models/donation_tax_receipt.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        string="Date", required=True, default=fields.Date.context_today, index=True
+    )
+    donation_date = fields.Date(string="Donation Date")
+    amount = fields.Monetary(string="Amount", currency_field="currency_id")
+    currency_id = fields.Many2one(
+        "res.currency",
+        string="Currency",
+        required=True,
+        ondelete="restrict",
+        default=lambda self: self.env.company.currency_id.id,
```
Comment (tbaden) [inline donation_base/models/donation_tax_receipt.py:27]: why was this change?
before:
`self.env.user.company_id.currency_id.id`

File: donation_base/security/tax_receipt_security.xml (hunk 1 above)
Comment (etobella) [reply donation_base/security/tax_receipt_security.xml:13]: Companies on 13 have no parent relation

File: donation_base/models/donation_tax_receipt.py (hunk 2 above)
Comment (etobella) [reply donation_base/models/donation_tax_receipt.py:27]: On 13 this is the way to get the current company

Comment (tbaden) [reply donation_base/models/donation_tax_receipt.py:27]: no it's still
`self.env.user.company_id.currency_id`
for multicompany you need the company_id of the user, which is the previous line

`self.env.company`
just gets the current env company

File: donation_base/security/tax_receipt_security.xml (hunk 1 above)
Comment (tbaden) [reply donation_base/security/tax_receipt_security.xml:13]: yes, there is
https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/res_company.py#L68

Comment (etobella) [reply donation_base/security/tax_receipt_security.xml:13]: But there is no parent path, so it shouldn't work

File: donation_base/models/donation_tax_receipt.py (hunk 2 above)
Comment (etobella) [reply donation_base/models/donation_tax_receipt.py:27]: Not according the migration guide
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0
https://www.odoo.com/documentation/13.0/developer/howtos/company.html

File: donation_base/security/tax_receipt_security.xml (hunk 1 above)
Comment (etobella) [reply donation_base/security/tax_receipt_security.xml:13]: Also, you can check this on the migration guide

Comment (tbaden) [reply donation_base/security/tax_receipt_security.xml:13]: it does also work without parent_path
https://github.com/odoo/odoo/blob/14.0/odoo/osv/expression.py#L526

and there is a difference between in, only including the actual ids, and child_of, actual_ids and their childs. so this would change this function.
not sure whats the authors original intend was, but in migration I would like to avoid a functional change

File: donation_base/models/donation_tax_receipt.py (hunk 2 above)
Comment (tbaden) [reply donation_base/models/donation_tax_receipt.py:27]: The inconsistency of odoo :)
https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/res_company.py#L194

they still using `self.env.user.company_id` themself.
as both are working I would preffer to use the previous one, as it makes it easier to find for possible patches and fixes later on as the code stays consistent

File: donation_base/security/tax_receipt_security.xml (hunk 1 above)
Comment (etobella) [reply donation_base/security/tax_receipt_security.xml:13]: In any case, the standard usage in 13 is to use in company_ids, so I think it is not an issue

File: donation_base/models/donation_tax_receipt.py (hunk 2 above)
Comment (etobella) [reply donation_base/models/donation_tax_receipt.py:27]: In your comment it is using user.company_ids, it is not the same

Comment (tbaden) [reply donation_base/models/donation_tax_receipt.py:27]: https://github.com/odoo/odoo/blob/13.0/addons/repair/models/repair.py#L312

File: donation_base/security/tax_receipt_security.xml (hunk 1 above)
Comment (tbaden) [reply donation_base/security/tax_receipt_security.xml:13]: it wasn't an issue for the beginning, so why change it?

File: donation_base/models/donation_tax_receipt.py (hunk 2 above)
Comment (etobella) [reply donation_base/models/donation_tax_receipt.py:27]: In any case, we should must env.company_id, as this one is the specified in the context, otherwise, it will use env.user.company_id.

https://github.com/odoo/odoo/blob/13.0/odoo/api.py#L525

allowed_company_ids is defined in the selector of the company. On the previous Odoo versions, multicompany was solved using the parent_id. Now it is solved using a multiselector on the header. So, this should be the way.

File: donation_base/security/tax_receipt_security.xml (hunk 1 above)
Comment (etobella) [reply donation_base/security/tax_receipt_security.xml:13]: Because it is the new behavior, company_ids will take the company that the user is seeing right now. So the right option is to use it this way. So, you might have selected to view 3 companies and, with the new logic, they will not be related. You must be able to see all the information at the same time.

The parent field is hidden and is no longer necessary to use it with this changes. All the logic on 13 (in Odoo and OCA) follows this behavior.

**Review by olgamarcocb — APPROVED** [review]
LGTM :+1:

**Review by LoisRForgeFlow — APPROVED** [review]
Technical review LGTM

**Comment by etobella** [comment]
@alexis-via Might be able to obtain your reviews here?

File: donation_base/models/donation_tax_receipt.py (hunk 2 above)
Comment (alexis-via) [reply donation_base/models/donation_tax_receipt.py:27]: I confirm that we must use self.env.company in v13 and not self.env.user.company_id. Obviously, the code of the official addons is far from perfect and Odoo itself hasn't achieved the full migration !

File: donation_base/security/tax_receipt_security.xml (hunk 1 above)
Comment (alexis-via) [reply donation_base/security/tax_receipt_security.xml:13]: I confirm that the change is good.

**Review by alexis-via — APPROVED** [review]
